diff -u -r a/src/shared/tpm2-util.c b/src/shared/tpm2-util.c
--- a/src/shared/tpm2-util.c	2025-09-03 14:35:40.000000000 -0400
+++ b/src/shared/tpm2-util.c	2025-10-09 22:15:15.565323489 -0400
@@ -5526,6 +5526,10 @@
                                                seal_key_handle);
 
                 primary_alg = primary_public->publicArea.type;
+                
+                // use attributes from sealing key to allow recoverable seal key
+                if (!(primary_public->publicArea.objectAttributes & TPMA_OBJECT_FIXEDTPM)) 
+                        hmac_template.objectAttributes = 0;           
         } else {
                 if (seal_key_handle != 0)
                         log_debug("Using primary alg sealing, but seal key handle also provided; ignoring seal key handle.");
